### PR TITLE
fix bucket.Create method description

### LINKED ISSUE #170: escrow bucket.Create does not save
In package `x/escrow`, [`Bucket.Create`](https://github.com/iov-one/weave/blob/master/x/escrow/model.go#L164) is described to be saving the created instance (see comment), but it only creates an ID and returns entity without storing it.

Method is used in [creation handler](https://github.com/iov-one/weave/blob/master/x/escrow/handler.go#L91) and it seems to expect the instance to be stored in the database when `Create` is called.

### PR DESCRIPTION
**Pull request by husio**
Escrow `Bucket.Create` method documentation is updated to be acurate.

fix #170

I think `Bucket.Create` method should be removed and assignment of an escrow ID moved to `Deposit` method. Current implementation goes through many indirect steps to achieve something that would best done in one place.

### CODE REVIEW

**Comment by husio** [comment]
Please have a look once more.

**Comment by ruseinov** [comment]
@husio except you need to fix a bunch of tests now

**Comment by husio** [comment]
I have renamed `Create` to `Build` instead. It would be very hard to change this due to validation and layered functionality.

**Comment by ruseinov** [comment]
@husio we are using create all over the place. If you look through the code - Build is used for something else, that has nothing to do with orm.

I'd like us to keep consistent in that regard instead of going the short way.

@ethanfrey would be the judge of whether we want to spend time here or not, I vote for not introducing new concepts in our code unless absolutely necessary.

**Comment by husio** [comment]
I agree with @ruseinov that it's good to be consistent.

Current `master` logic is hard to change, because the controller is expecting an Escrow instance with and ID and zero funds. Only bucket can assign an ID, but it cannot save it with zero funds because validation does not allow for that. I think validation is correct, because zero escrow does not make sense.
Bucket cannot create an escrow with funds, because this logic is reserved for controller.

To summarize, the controller requires a zero amonut escrow with an ID, but validation does not allow for such entity to be persisted.

Changing any part of this logic (and current implementation is not consistent) requires update to many layers and will be a bigger change.

**Comment by ruseinov** [comment]
@husio let's leave it at that, so we cannot really save, because of the validation. 
I'd say it's good to go, will merge this.
